Automattic/wp-calypso
# Reader Recommendations: dismiss the site rather than blocking it

Fixes https://github.com/Automattic/wp-calypso/issues/20392.

When a user dismisses a site recommendation, we currently block the site. As @jancavan rightly says in the issue:

> Dismissing doesn't mean "blocking"; it just means "don't recommend this to me again."

This PR switches to use the `/me/dismiss/sites/:blog_id/new` endpoint in this scenario instead.

Todo:
- [x] switch to use dismissSite action
- [x] stored dismissed sites in Redux
- [x] add `getDismissedSites` selector
- [x] switch requests to data layer
- [x] move actions under site-dismissals

### To test

In Following Manage http://calypso.localhost:3000/following/manage, locate the site recommendations underneath the search box:

<img width="803" alt="screen shot 2018-07-16 at 18 15 29" src="https://user-images.githubusercontent.com/17325/42745031-a85c78a4-8924-11e8-8c1c-9a0881ae15fe.png">

Try dismissing the recommendation using the 'X'. Make sure that you see an API call to the dismiss site endpoint and that the rec disappears to be replaced with a new one.

Repeat the process with in-stream recommendations (http://calypso.localhost:3000):

<img width="975" alt="screen shot 2018-07-16 at 18 16 23" src="https://user-images.githubusercontent.com/17325/42745075-e42b92d4-8924-11e8-9350-8e1cba77839a.png">
